src/diracx/routers/job_manager/__init__.py

```diff
@@ -406,6 +496,98 @@ async def get_single_job(job_id: int):
     return f"This job {job_id}"
 
 
+@router.delete("/{job_id}")
+async def delete_single_job(
+    job_id: int,
+    config: Annotated[Config, Depends(ConfigSource.create)],
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    """
+    Delete a job by killing and setting the job status to DELETED.
+    """
+
+    # TODO: implement job policy
+    try:
+        await delete_job(
+            job_id,
+            config,
+            job_db,
+            job_logging_db,
+            task_queue_db,
+            background_task,
+        )
+    except NoResultFound as e:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND.value, detail=str(e)
+        ) from e
+
+    return f"Job {job_id} has been successfully deleted"
+
+
+@router.post("/{job_id}/kill")
+async def kill_single_job(
+    job_id: int,
+    config: Annotated[Config, Depends(ConfigSource.create)],
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    """
+    Kill a job.
+    """
+
+    # TODO: implement job policy
+
+    try:
+        await kill_job(
+            job_id, config, job_db, job_logging_db, task_queue_db, background_task
+        )
+    except NoResultFound as e:
+        raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail=str(e)) from e
+
+    return f"Job {job_id} has been successfully killed"
+
+
+@router.post("/{job_id}/remove")
+async def remove_single_job(
+    job_id: int,
+    config: Annotated[Config, Depends(ConfigSource.create)],
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    sandbox_metadata_db: SandboxMetadataDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    """
+    Fully remove a job from the WMS databases.
+
+    WARNING: This endpoint has been implemented for the compatibility with the legacy DIRAC WMS
+    and the JobCleaningAgent. However, once this agent is ported to diracx, this endpoint should
+    be removed, and the delete endpoint should be used instead.
+    """
+
```

Comment:
So it can be more easily found by grepping:
```suggestion
    # TODO: Remove once legacy DIRAC no longer needs this
```